src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs

```diff
@@ -95,5 +95,39 @@ public void MulticastOption_CreateSocketSetGetOption_GroupAndInterfaceIndex_SetS
                 Assert.Throws<SocketException>(() => socket.GetSocketOption(SocketOptionLevel.IP, SocketOptionName.AddMembership));
             }
         }
+
+        [Fact]
+        public void MulticastInterface_SetToInterfaceIndexAnySucceeds()
+        {
+            IPAddress ip = IPAddress.Parse ("239.255.255.250");
+            int index = 0;
+            using (Socket s = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                s.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.MulticastInterface, IPAddress.HostToNetworkOrder(index));
+            }
+        }
+
+        [Fact]
+        public void MulticastInterface_SetToInterfaceIndexLoopbackSucceeds()
+        {
+            IPAddress ip = IPAddress.Parse ("239.255.255.250");
+            int index = 1;
+            using (Socket s = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                s.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.MulticastInterface, IPAddress.HostToNetworkOrder(index));
+            }
+        }
+
+        [Fact]
+        public void MulticastInterface_SetToInterfaceIndexInvalidThrows()
+        {
+            IPAddress ip = IPAddress.Parse ("239.255.255.250");
+            int index = 31415;
+            using (Socket s = new Socket(AddressFamily.InterNetwork, SocketType.Dgram, ProtocolType.Udp))
+            {
+                Assert.Throws<SocketException>(() =>
+                    s.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.MulticastInterface, IPAddress.HostToNetworkOrder(index)));
+            }
+        }
```

Comment:
Is there any test we could write to validate that SetSocketOption not only returned successfully but also that that the option had the intended affect on behavior?